spec/rspec/core/suite_hooks_spec.rb

```diff
@@ -23,6 +23,13 @@ module RSpec::Core
           }.to raise_error(ZeroDivisionError)
         end
 
+        it 'runs in the context of an example group' do
+          run_context = nil
+          RSpec.configuration.__send__(registration_method, :suite) { run_context = self }
+          RSpec.configuration.with_suite_hooks { }
+          expect(run_context).to be_a ExampleGroup
+        end
```

Comment:
I think it's worth adding some lines to this (or perhaps making a new spec) showing that rspec-mocks and rspec-expectations APIs can be used from within the hook.  Ultimately that's the bug we're trying to fix.  l think that as rspec-core currently works, any block evaluated in the context of an `ExampleGroup` instance would support those APIs but there's no guarantee that will always be true.